### PR TITLE
Move Configuration into Abstractions Project

### PR DESCRIPTION
**Pull request by shep1987**
Based on feedback in https://github.com/NuKeeperDotNet/NuKeeper/issues/489
I think a possible first step is as follows:

- Setup Nukeeper.Abstractions Project
- Add Code Rules
- Move Configuration not dependent on Octokit into abstractions project
- GitHubAuthSettings  => AuthSettings, URI and Token seem like sensible props for now

Hopefully this should give everyone wanting to help a base to start from.

Shaun

### CODE REVIEW

**Comment by AnthonySteele** [comment]
Looks good so far! Will merge tomorrow morning.
For future reference,  a couple of files appear to have white-space changes. But the combo of `.editorconfig` and `.gitattributes`should prevent that?

**Comment by shep1987** [comment]
Thanks!

Visual studio seemed insistent on “fixing” those files am happy to revert the white space changes if you prefer?

File: NuKeeper/Commands/MultipleRepositoryCommand.cs [hunk 1]
```diff
@@ -1,103 +1,104 @@
...
-            catch (Exception ex)
-            {
-                return ValidationResult.Failure($"Unable to parse regex '{value}' for ExcludeRepos: {ex.Message}");
-            }
-
-            return ValidationResult.Success;
-        }
-    }
-}
+using System;
```
Comment (MarcBruins) [inline NuKeeper/Commands/MultipleRepositoryCommand.cs]: Did anything happen here?

Comment (shep1987) [reply NuKeeper/Commands/MultipleRepositoryCommand.cs]: A using statement was added is the only functional change

There was a space at the end of each line which Visual Studio wanted to remove. As above am happy to revert this if you like.

This occurred in 
NuKeeper.Tests\Configuration\FileSettingsReaderTests.cs
NuKeeper\Commands\MultipleRepositoryCommand.cs
NuKeeper.Abstractions\Configuration\FileSettings.cs

image

**Pull request merged**